essential-documentation/contribute-to-appflowy/write-for-appflowy/drafts/draft-how-to-add-a-new-property-to-appflowy-database.md

```diff
@@ -1,2 +1,92 @@
-# \[Draft] How to add a new property to appflowy database
+One of the things you can do in Appflowy is work with structured data (it's called Appflowy databases) and view them in different layouts, such as Table, Board, Calendar.
 
+![flow-database.png](assets/flowy-database.png)
+In this article, I will take a look at how the back-end part of Appflowy databases works overal and a more detailed guide on how to create a new property type.
+## Flowy-database
+Flowy-database is a rust crate mainly responsible for the abstractions behind Appflowy databases, CRUD actions on data, storing, filtering and etc. Let's take a look at the code structure and dive in (the picture is only the directories you can take a more detailed look [here](https://github.com/AppFlowy-IO/AppFlowy/tree/e0ad364fa3e9da183698739e0661beef22b1728d/frontend/rust-lib/flowy-database2/src)).
+
+![flowy-database-code-structure.png](assets/flowy-database-code-structure.png)
+### [Entities](https://github.com/AppFlowy-IO/AppFlowy/tree/e0ad364fa3e9da183698739e0661beef22b1728d/frontend/rust-lib/flowy-database2/src/entities)
```

Comment:
For readability sake of the markdown document, maybe some newlines before new headings.

Reply:
At least in github preview, it seems to ignore those new lines. But I added it. Maybe it will be different on gitbook.